File: dehaze-python/app/models/entity/sys_file.py
"""
文件相关实体模型
"""

from typing import Optional

from app.database import Base
from app.models.base import BaseModel
from sqlalchemy import (CHAR, BigInteger, Index, Integer,
                        String)
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column


class SysFile(BaseModel):
    __tablename__ = 'sys_file'
    __table_args__ = (
        Index('md5_key', 'md5', unique=True),
        {'comment': '文件表'}
    )

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment='文件id')
    type: Mapped[Optional[str]] = mapped_column(
        String(100), nullable=True, comment='文件类型')
    name: Mapped[str] = mapped_column(
        String(100), nullable=False, comment='文件原始名')
    object_name: Mapped[str] = mapped_column(
        String(100), nullable=False, comment='对象键（存储后端中的定位，与环境无关）')
    storage: Mapped[str] = mapped_column(
        String(32), nullable=False, default='minio',
        comment='存储后端标识(minio/local/nginx-static)')
    size: Mapped[str] = mapped_column(
        String(100), nullable=False, default='0', comment='文件大小(格式化)')
    size_bytes: Mapped[Optional[int]] = mapped_column(
        Integer, nullable=True, default=0, comment='文件大小(字节)')
    md5: Mapped[str] = mapped_column(CHAR(32), nullable=False, unique=True,
                                     comment='文件的MD5值，用于比对文件是否相同')
    deleted: Mapped[int] = mapped_column(
        mysql_types.TINYINT, nullable=False, default=0, comment='逻辑删除标识(0:未删除;1:已删除)')

File: dehaze-go/internal/model/sys_file.go
package model

// SysFile 文件表
// URL 永远运行时拼接（storage.baseUrl + object_name），不落库
type SysFile struct {
	BaseModel
	Type       *string `gorm:"column:type;type:varchar(100);comment:文件类型" json:"type"`
	Name       string  `gorm:"column:name;type:varchar(100);not null;comment:文件原始名" json:"name"`
	ObjectName string  `gorm:"column:object_name;type:varchar(100);not null;comment:对象键（存储后端中的定位，与环境无关）" json:"objectName"`
	Storage    string  `gorm:"column:storage;type:varchar(32);not null;default:minio;comment:存储后端标识(minio/local/nginx-static)" json:"storage"`
	Size       string  `gorm:"column:size;type:varchar(100);not null;default:'0';comment:文件大小（格式化显示）" json:"size"`
	SizeBytes  *int64  `gorm:"column:size_bytes;type:bigint;comment:文件大小（原始字节数）" json:"sizeBytes"`
	MD5        string  `gorm:"column:md5;type:char(32);not null;uniqueIndex:uk_md5;comment:文件的MD5值，用于比对文件是否相同" json:"md5"`
	Deleted    int8    `gorm:"column:deleted;type:tinyint;not null;default:0;comment:逻辑删除标识(0:未删除;1:已删除)" json:"deleted"`
}

func (SysFile) TableName() string {
	return "sys_file"
}
